Remove Outdated Optnone Workaround for LLVM

LumioseSil opened the pull request:
#### fff341fae4fa8e899f5f552416a3b8e1d234b2f7
<pre>
Remove Outdated Optnone Workaround for LLVM
<a href="https://bugs.webkit.org/show_bug.cgi?id=252315">https://bugs.webkit.org/show_bug.cgi?id=252315</a>
&lt;rdar://problem/49792205&gt;

Reviewed by NOBODY (OOPS!).

There is no more need for the optnone attribute now that LLVM can
compile arm64_32 assembler properly.

*Source/JavaScriptCore/runtime/CachedTypes.cpp: Remove optnone attribute
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/33a908d5c964486846b8c5d8cf7ab8038ce1461f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/687 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/706 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/734 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/887 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/605 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/755 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/791 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/840 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/694 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/660 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/644 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/857 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/733 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/639 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/670 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/604 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/618 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/660 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/679 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/650 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/641 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/710 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/603 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/645 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/148 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/660 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/711 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/142 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

Constellation reviewed (changes requested):
r- based on previous discussion.
CC @geoffreygaren

Constellation reviewed (changes requested):
Nothing is changed in this PR.

LumioseSil committed:
Remove Outdated Optnone Workaround for LLVM
https://bugs.webkit.org/show_bug.cgi?id=252315
<rdar://problem/49792205>

Reviewed by NOBODY (OOPS!).

There is no more need for the optnone attribute now that LLVM can
compile arm64_32 assembler properly.

*Source/JavaScriptCore/runtime/CachedTypes.cpp: Remove optnone attribute
